pkg/vetter/serviceportprefix/vet.go

```diff
@@ -54,19 +58,24 @@ func (m *SvcPortPrefix) Vet() ([]*apiv1.Note, error) {
 		return nil, err
 	}
 	for _, s := range services {
+		var unsupportedPortPrefixes []string
 		for _, p := range s.Spec.Ports {
 			if p.Protocol != util.ServiceProtocolUDP &&
 				util.ServicePortPrefixed(p.Name) == false {
-				notes = append(notes, &apiv1.Note{
-					Type:    servicePortPrefixNoteType,
-					Summary: servicePortPrefixSummary,
-					Msg:     servicePortPrefixMsg,
-					Level:   apiv1.NoteLevel_WARNING,
-					Attr: map[string]string{
-						"service_name": s.Name,
-						"namespace":    s.Namespace}})
+				unsupportedPortPrefixes = append(unsupportedPortPrefixes, p.Name)
 			}
 		}
+		if unsupportedPortPrefixes != nil && len(unsupportedPortPrefixes) > 0 {
```

Comment:
It might be ok to remove one check and just do `len(unsupportedPortPrefixes) > 0`